solc/CommandLineParser.cpp

```diff
@@ -1276,7 +1281,7 @@ void CommandLineParser::processArgs()
 		// switch to assembly mode
 		using Input = yul::YulStack::Language;
 		using Machine = yul::YulStack::Machine;
-		m_options.assembly.inputLanguage = m_args.count(g_strYul) ? Input::Yul : (m_args.count(g_strStrictAssembly) ? Input::StrictAssembly : Input::Assembly);
+		m_options.assembly.inputLanguage = m_args.count(g_strStrictAssembly) ? Input::StrictAssembly : Input::Assembly;
```

Comment:
If we decided it's ok to remove `--yul` in a non-breaking version, why not go all the way and remove `--assemble` as well? It's also useless and confusing.

As far as I can tell it has been neutered ages ago and made equivalent to `--strict-assembly`, but it's not obvious looking at how it's handled here or at `--help`.

A few years ago I needed to understand how these dialect/machine/language options really work and made a table summarizing their combinations: https://github.com/ethereum/solidity/issues/7830#issuecomment-718227359. Though now I see that even back then I fell into this trap and wrongly  assumed that `Input::Assembly` was a distinct option (because otherwise why bother handling it like this?) but apparently not because [`languageToDialect()`](https://github.com/ethereum/solidity/blob/v0.8.26/libyul/YulStack.cpp#L50-L62) collapses them into the same thing.